subsys/net/lib/lwm2m/lwm2m_obj_firmware_pull.c

```diff
@@ -450,7 +450,7 @@ int lwm2m_firmware_start_transfer(char *package_uri)
 	/* close old socket */
 	if (firmware_ctx.sock_fd > 0) {
 		lwm2m_socket_del(&firmware_ctx);
-		close(firmware_ctx.sock_fd);
+		(void)close(firmware_ctx.sock_fd);
```

Comment:
I understand that this addresses the coverity issue of ignored non-void return, but, I am wondering, if this should actually be checked for some error. I leave this to @jukkar and the other networking folks.

Reply:
Today is a holiday in the US but I can take some time tomorrow and determine what to do with return code here.

Reply:
I wonder if there is actually anything to do. AFAIK the only case the `close` call can fail currently is when we provide invalid socket fd. And I don't see a way to recover from this other than ignore the corrupted socket.

Posix seems to allow also `EINTR`, but I didn't see it implemented, nor I'm not sure how should it be handled (retry?). I've not seen such an advanced `close` error management implemented anywhere in Zephyr.

Optionally we could log this event or even assert on it, I've just went straightforward with this patch.

Reply:
This is similar case as in #16696. We need to ask ourselves that if the close fails, what would the caller do in that case. Normally there is not much to do here, the socket should not be used anyway so ignoring the return value is fine.

Reply:
Yep, I'm good with this change as it seems to be what we are doing for the moment.